addons/mail/static/src/components/composer_text_input/composer_text_input.js

```diff
@@ -241,18 +241,41 @@ class ComposerTextInput extends Component {
     _onKeydownTextareaEnter(ev) {
         if (this.composer.hasSuggestedPartners) {
             ev.preventDefault();
-        } else {
-            if (!this.props.hasSendOnEnterEnabled) {
-                return;
-            }
-            if (ev.shiftKey) {
-                return;
-            }
-            if (this.env.messaging.device.isMobile) {
-                return;
-            }
+            return;
+        }
+        if (
+            this.props.sendShortcuts.includes('ctrl-enter') &&
+            !ev.altKey &&
+            ev.ctrlKey &&
+            !ev.metaKey &&
+            !ev.shiftKey
+        ) {
             this.trigger('o-keydown-enter');
             ev.preventDefault();
+            return;
+        }
+        if (
+            this.props.sendShortcuts.includes('enter') &&
+            !ev.altKey &&
+            !ev.ctrlKey &&
+            !ev.metaKey &&
+            !ev.shiftKey &&
+            !this.env.messaging.device.isMobile
+        ) {
+            this.trigger('o-keydown-enter');
+            ev.preventDefault();
+            return;
+        }
+        if (
+            this.props.sendShortcuts.includes('meta-enter') &&
+            !ev.altKey &&
+            !ev.ctrlKey &&
+            ev.metaKey &&
+            !ev.shiftKey
+        ) {
+            this.trigger('o-keydown-enter');
```

Comment:
Do we really need a trigger for this? And if we must, the name is incorrect. It should be renamed to something "send message" if that's its purpose (or have one trigger name per key combination at worse), but it should not be called just "enter" when it was actually a different key combination.

Reply:
indeed, ideally want to get rid of this event, but in context of PR I think I can at least improve its name.